waspc/src/Wasp/AppSpec/Valid.hs

```diff
@@ -84,10 +87,14 @@ validateWaspVersion specWaspVersionStr = eitherUnitToErrorList $ do
     incompatibleVersionError :: SV.Version -> SV.Range -> ValidationError
     incompatibleVersionError actualVersion expectedVersionRange =
       GenericValidationError $
-        unwords
+        unlines
           [ "Your Wasp version does not match the app's requirements.",
             "You are running Wasp " ++ show actualVersion ++ ".",
-            "This app requires Wasp " ++ show expectedVersionRange ++ "."
+            "This app requires Wasp " ++ show expectedVersionRange ++ ".",
+            "To install specific version of Wasp, do:",
+            "  curl -sSL https://get.wasp-lang.dev/installer.sh | sh -s -- -v x.y.z",
```

Comment:
Since our installer expects 4 parts, should we show them an extra zero or place somewhere?

Also, do we want to just construct the minimal version for them in the example? So in my case of `This app requires Wasp ^0.7.0.` we could say: `For example: curl -sSL https://get.wasp-lang.dev/installer.sh | sh -s -- -v 0.7.0.0` I think once our semantic version matches the installer (e.g. 3 parts) it will be easier, but now it may be confusing for them. Just a thought.

Reply:
Ok so that is something that happened in another PR from a contributor couple of days ago -> we switched to x.y.z! That is already in main actually. So while the current version is 0.6.0.0, the next one will be 0.7.0, as per semver rules for alpha releases, where basically there is no "patch" field, it is just 0.major.minor . Ok actually it is 0.minor.patch, but minor is treated as major and patch is treated as minor.
So that should cover most of your question, but the part that remains is the idea of offering them immediately which version to install. That is a bit tricky because I have a semver range here, so I would need to pick a version from it, but I am not sure which version from that range exists or not. We could maybe assume it is the start of the range, but does not have to be true in theory.

We will anyway be upgrading the whole system of upgrading and picking versions in the future I think, possibly make it automatic based on the project and similar, so that is why I  wasn't super worried about optimizing this completely.
I will be writing more about that in another PR soon!

Reply:
Ok thanks, the 3 fields make sense. Yeah, the picking a version for them is hard if we want to pick the latest, or some specific version in the range. But I think we could just pick the lowest semver version for them and assume it will work since it was in the wasp file, right? But this is not a big deal, let's just punt on all that for now. 👍🏻